src/Stryker.Core/Stryker.Core/MutantFilters/SinceMutantFilter.cs

```diff
@@ -74,12 +93,30 @@ public IEnumerable<IMutant> FilterMutants(IEnumerable<IMutant> mutants, IReadOnl
 
         // If any of the tests have been changed, we want to return all mutants covered by these testfiles.
         // Only check for changed c# files. Other files have already been handled.
-        if (_diffResult.ChangedTestFiles != null && _diffResult.ChangedTestFiles.Any(file => file.EndsWith(".cs")))
+        if (_diffResult.ChangedTestFiles != null && _diffResult.ChangedTestFiles.Keys.Any(file => file.EndsWith(".cs")))
         {
-            filteredMutants = ResetMutantStatusForChangedTests(mutants);
+            foreach (var mutant in mutants)
+            {
+                foreach (var line in _diffResult.ChangedTestFiles[file.FullPath])
+                {
+                    var actualLineSpan = mutant.Mutation.OriginalNode.GetLocation().GetMappedLineSpan();
+
+                    _logger.LogDebug("Original line span is found to be {actualLineSpan}", actualLineSpan);
+                    var start = actualLineSpan.Span.Start.Line;
+                    var end = actualLineSpan.Span.Start.Line;
+                    if (start <= line && line <= end)
+                    {
+                        if (mutant.ResultStatus != MutantStatus.NoCoverage)
+                        {
+                            mutant.ResultStatus = MutantStatus.Pending;
+                            mutant.ResultStatusReason = "Mutant changed compared to target commit";
```

Comment:
The status reason is not correct, the mutant has not been changed but a covering test has been changed